daemon/graphdriver/driver.go

```diff
@@ -142,10 +142,36 @@ func GetDriver(name, home string, options []string) (Driver, error) {
 func New(root string, options []string) (driver Driver, err error) {
 	for _, name := range []string{os.Getenv("DOCKER_DRIVER"), DefaultDriver} {
 		if name != "" {
+			logrus.Infof("[graphdriver] trying provided driver %q", name) // so the logs show specified driver
 			return GetDriver(name, root, options)
 		}
 	}
 
+	// Guess for prior driver
```

Comment:
Basically, I think most users will select a driver and "stick" with it. It's okay for Docker to pick the best available default for new installs, but when "in doubt", the user should be in full control. Adding a `-s` flag shouldn't be too much of a trouble.

A hard fail is easier to detect and Google than some warning that was briefly shown when starting the daemon.

Those that like to switch drivers on a daily base are probably experienced enough to understand what needs to be done and (IMO) not the main audience.